### PR TITLE
CFY-4442 updated plugins docs

### PR DESCRIPTION
**Pull request by nirbir**

### CODE REVIEW

File: content/apis/plugins-common.html [hunk 1]
```diff
@@ -10,6 +10,6 @@
 <html>
   <head>
     <meta http-equiv="content-type" content="text/html; charset=utf-8"/>
-    <meta http-equiv="refresh" content="0;url=http://cloudify-plugins-common.readthedocs.org/en/3.2/"/>
+    <meta http-equiv="refresh" content="0;url=http://cloudify-plugins-common.readthedocs.org/en/3.3/"/>
```
Comment (nirbir) [inline content/apis/plugins-common.html]: Link is broken. Any other link?
Remove reference to it?

Comment (nir0s) [inline content/apis/plugins-common.html]: no no.. we need to deploy this to readthedocs

File: content/plugins/overview.md [hunk 2]
```diff
@@ -17,60 +17,9 @@ To cover the two major types of plugins (IaaS and Management tools), let's also
 
 # Plugin Development
 
-You should check out the [Plugins Authoring Guide](plugins-authoring.html) if you want to write your own plugin for your chosen tool or IaaS provider.
+You should check out the [Plugins Authoring Guide]({{< relref "plugins/creating-your-own-plugin.md" >}}) if you want to write your own plugin for your chosen tool or IaaS provider.
```
Comment (nir0s) [inline content/plugins/overview.md]: Please don't call it the "Plugin Authoring Guide" :)
Maybe: You should check out the "Creating your own plugin" page if...
?

File: content/plugins/overview.md [hunk 3]
```diff
@@ -17,60 +17,9 @@ To cover the two major types of plugins (IaaS and Management tools), let's also
...
 
-The Python module which provides the API for a plugin to interact with Cloudify is called the cloudify-plugins-common module.
-The module provides features for getting and setting context, downloading blueprint resources and much more and its reference can be found [here](apis-plugins-common.html).
+In order to learn how to create and use a plugin package, see the [plugins usage guide]({{< relref "plugins/using-plugins.md" >}})
```
Comment (nir0s) [inline content/plugins/overview.md]: Again, please let's not name this "The X Guide"..

maybe.. see the "using plugins" page? :)

File: content/plugins/using-plugins.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
+---
+layout: bt_wiki
+title: Working With Plugins
+category: Docs
+draft: false
+weight: 50
+
+---
+
+The Cloudify manager exposes a plugins API, used to allow the installation of Python dependencies from pre-packed Python wheels. The plugins could come in handy for the following:
```
Comment (nir0s) [inline content/plugins/using-plugins.md]: The Cloudify manager -> Cloudify Manager

File: content/plugins/using-plugins.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
+---
+layout: bt_wiki
+title: Working With Plugins
+category: Docs
+draft: false
+weight: 50
+
+---
+
+Cloudify Manager exposes a plugins API, used to allow the installation of Python dependencies from pre-packed Python wheels. The plugins could come in handy for the following:
```
Comment (nir0s) [inline content/plugins/using-plugins.md]: pre-packaged Python wheels

File: content/plugins/using-plugins.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+category: Docs
+draft: false
+weight: 50
+
+---
+
+Cloudify Manager exposes a plugins API, used to allow the installation of Python dependencies from pre-packaged Python wheels. The plugins could come in handy for the following:
+
+* Allowing installation of Python dependencies without needing access to PyPi or the internet.
+* Eliminating the need for system level compilation tools such as gcc.
```
Comment (nir0s) [inline content/plugins/using-plugins.md]: so.. these two are simply not the point. The point is that you can use the plugins API to install Cloudify plugins.
The fact that they don't require post-compilation is irrelevant at the point and should be noted as a side note later on. Also, the internet thing is, once again, irrelevant. The fact that we needed a connection up until now was a fluke. Not needing access to the internet is not a feature.

The point was need to make here (including the first line) is:

Cloudify Manager exposes a plugins API used to allow users to upload Cloudify plugins to Cloudify Manager and install them.

Please add the notes regarding the no internet connection and no compilers needed to the end.

thanks

File: content/plugins/using-plugins.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+### Creating A Plugin Package
+
+The official Cloudify tool used to create plugin packages is named [Wagon](https://github.com/cloudify-cosmo/wagon). In addition to the required package wheels, plugin packages created by Wagon also contain metadata regarding the plugin such as package name, compiled distribution, plugin package version and more.
```
Comment (nir0s) [inline content/plugins/using-plugins.md]: package wheels -> wheels

Comment (nir0s) [inline content/plugins/using-plugins.md]: compiled distribution -> the distribution the plugin was compiled on (if not Pure Python)

File: content/plugins/using-plugins.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+The plugin metadata will be used to determine the compatibility of the plugin upon installation on the destination host. Wagon allows creating packages using PyPi or using the actual plugin source code, for example:
```
Comment (nir0s) [inline content/plugins/using-plugins.md]: plugin metadata -> plugin's metadata

File: content/plugins/using-plugins.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+{{< gsHighlight  bash  >}}
+$> cd /path/to/python-example-package
```
Comment (nir0s) [inline content/plugins/using-plugins.md]: /path/to/plugin/root/dir

File: content/plugins/using-plugins.md [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+   INFO - Removing previous archive...
+   INFO - Creating tar.gz archive: ./python_example_package-1.0-py27-none-linux_x86_64-Ubuntu-trusty.wgn...
+   INFO - Process complete!
+
+{{< /gsHighlight >}}
+To learn more about how to create a plugin package, see the [Wagon documentation](https://github.com/cloudify-cosmo/wagon).
+
+### Using Plugins
+
+The Cloudify RESTful API enables the upload, download, delete and list of all plugins stored on the Cloudify manager. These abilities are available from the rest client as well as via the CLI. For example:
```
Comment (nir0s) [inline content/plugins/using-plugins.md]: The Cloudify -> Cloudify's
upload -> uploading
download -> downloading
delete -> deletion
list -> listing

File: content/plugins/using-plugins.md [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+{{< /gsHighlight >}}
+
+After uploading the relevant plugin, blueprints can make use of it by having the plugin defined in the blueprint itself.
+
+
+{{% gsNote title="Note" %}}
+Read more about how to define the plugin in the blueprint [here]({{< relref "blueprints/spec-plugins.md" >}}).
+{{% /gsNote %}}
+
+{{% gsTip title="Uploading plugins via manager blueprint" %}}
```
Comment (nir0s) [inline content/plugins/using-plugins.md]: Manager blueprint is an abstraction which should be disguised.

I suggest "Uploading plugins during bootstrap" and then "Cloudify enables uploading plugins to the Manager during bootstrap.

File: content/plugins/using-plugins.md [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+title: Working With Plugins
+category: Docs
+draft: false
+weight: 50
+
+---
+
+Cloudify Manager exposes a plugins API used to allow users to upload Cloudify plugins to Cloudify Manager and install them.
+
+* Allowing installation of Python dependencies without needing access to PyPi or the internet.
```
Comment (nir0s) [inline content/plugins/using-plugins.md]: ?

File: content/plugins/using-plugins.md [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+   INFO - Removing previous archive...
+   INFO - Creating tar.gz archive: ./python_example_package-1.0-py27-none-linux_x86_64-Ubuntu-trusty.wgn...
+   INFO - Process complete!
+
+{{< /gsHighlight >}}
+To learn more about how to create a plugin package, see the [Wagon documentation](https://github.com/cloudify-cosmo/wagon).
+
+### Using Plugins
+
+Cloudify's RESTful API enables the uploading, downloading, deletion and listing of all plugins stored on the Manager. These abilities are available from the rest client as well as via the CLI. For example:
```
Comment (nir0s) [inline content/plugins/using-plugins.md]: are available from -> are exposed by

File: content/plugins/using-plugins.md (hunk 12 above)
Comment (nir0s) [inline content/plugins/using-plugins.md]: oh.. I see that you added these as notes below. Please remove from here then.

File: content/plugins/using-plugins.md [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+{{% gsNote title="Note" %}}
+Read more about how to define the plugin in the blueprint [here]({{< relref "blueprints/spec-plugins.md" >}}).
+{{% /gsNote %}}
+
+{{% gsNote title="Note" %}}
+Pre-packaged plugins that were uploaded to Cloudify Manager eliminate the need for access to PyPI or the internet, and for system level compilation tools (such as gcc) during installation.
+{{% /gsNote %}}
+
+{{% gsTip title="Uploading plugins during bootstrap" %}}
+Cloudify enables uploading plugins uploading plugins to the Manager during bootstrap. For more on that, please refer to [Plugin Resources]({{< relref "blueprints/upload-resources.md" >}}#plugin-resources).
```
Comment (nir0s) [inline content/plugins/using-plugins.md]: uploading plugins X2

**Pull request merged**